### ISSUE TITLE
Error on saveAsset (v7.6.0) [iOS]

### ISSUE THREAD

**Issue by gezquinndesign** [react-native-cameraroll/react-native-cameraroll#615]
We're getting the following error on iOS when upgrading from 7.4.0 to the latest version (7.6.0):

<details>
<summary>Error</summary>

```js
{
  "code": "E_UNABLE_TO_SAVE",
  "message": "The operation couldn’t be completed. (PHPhotosErrorDomain error -1.)",
  "domain": "PHPhotosErrorDomain",
  "userInfo": {},
  "nativeStackIOS": [
    "0   Elsewhere Beta                      0x0000000100c48be8 RCTJSErrorFromCodeMessageAndNSError + 112",
    "1   Elsewhere Beta                      0x0000000100be56c4 __41-[RCTModuleMethod processMethodSignature]_block_invoke_2.73 + 152",
    "2   Elsewhere Beta                      0x0000000101326428 __57-[RNCCameraRoll saveToCameraRoll:options:resolve:reject:]_block_invoke.70 + 916",
    "3   Photos                              0x00000001af2d0454 72B67AD1-90CA-3201-A098-8F925C71C3B9 + 971860",
    "4   Photos                              0x00000001af2d1fdc 72B67AD1-90CA-3201-A098-8F925C71C3B9 + 978908",
    "5   Photos                              0x00000001af2d1a00 72B67AD1-90CA-3201-A098-8F925C71C3B9 + 977408",
    "6   Photos                              0x00000001af2d15f8 72B67AD1-90CA-3201-A098-8F925C71C3B9 + 976376",
    "7   CoreFoundation                      0x00000001979fbc84 3A5F992A-D1CD-312E-BD2E-F7C66343A417 + 203908",
    "8   CoreFoundation                      0x00000001979fb710 3A5F992A-D1CD-312E-BD2E-F7C66343A417 + 202512",
    "9   Foundation                          0x00000001968c6698 D92E19C1-6299-3E94-8614-C505D5ABCCDB + 415384",
    "10  Foundation                          0x00000001968c6218 D92E19C1-6299-3E94-8614-C505D5ABCCDB + 414232",
    "11  Foundation                          0x00000001968c5b80 D92E19C1-6299-3E94-8614-C505D5ABCCDB + 412544",
    "12  libxpc.dylib                        0x00000001f3a0dff8 042EA33B-3C8F-3FB8-869C-369A3115977C + 126968",
    "13  libxpc.dylib                        0x00000001f3a007a8 042EA33B-3C8F-3FB8-869C-369A3115977C + 71592",
    "14  libdispatch.dylib                   0x000000019f8c6e54 B7D9D0A6-2414-37F2-9D03-1A25A474E2CF + 15956",
    "15  libdispatch.dylib                   0x000000019f8e4740 B7D9D0A6-2414-37F2-9D03-1A25A474E2CF + 137024",
    "16  libdispatch.dylib                   0x000000019f8ce284 B7D9D0A6-2414-37F2-9D03-1A25A474E2CF + 45700",
    "17  libdispatch.dylib                   0x000000019f8cef64 B7D9D0A6-2414-37F2-9D03-1A25A474E2CF + 48996",
    "18  libdispatch.dylib                   0x000000019f8d9cb4 B7D9D0A6-2414-37F2-9D03-1A25A474E2CF + 93364",
    "19  libdispatch.dylib                   0x000000019f8d9528 B7D9D0A6-2414-37F2-9D03-1A25A474E2CF + 91432",
    "20  libsystem_pthread.dylib             0x00000001f39acf20 _pthread_wqthread + 288",
    "21  libsystem_pthread.dylib             0x00000001f39acfc0 start_wqthread + 8"
  ]
}
```
</details>

This comes from the following code:
```js
CameraRoll.saveAsset(pathToImage, { type: 'photo' })
```

where `pathToImage` is a cloudinary URL for a PNG image. We'd like to get 7.6.0 asap so we can get the privacy manifest included in our build.  Any ideas what might be happening?

**Comment by Arjan-Zuidema** [comment]
I recommend downloading the file to disk first, preferably the Documents directory.

**Comment by gezquinndesign** [comment]
@Arjan-Zuidema We were under the impression that `save` and `saveAsset` accept remote library URIs when not being called on Android?

**Comment by Arjan-Zuidema** [comment]
I think it should. But in my experience it is better to save to disk first

**Comment by gezquinndesign** [comment]
Ah, OK, thanks.  I guess you'd have to delete the image from Documents after saving to the camera roll? 

Though, I would have expected everything to be backwards compatible with minor version changes (7.4.0 works with no issues).

Incidentally, we switched from `save` to `saveAsset` because the typescript typings say it's now [deprecated](https://github.com/react-native-cameraroll/react-native-cameraroll/blob/74fb19f1d7db1e4d5e1edd43c3e92fe51bf7ab45/src/CameraRoll.ts#L244). Wondering if this may also be related.

**Comment by Arjan-Zuidema** [comment]
> Ah, OK, thanks. I guess you'd have to delete the image from Documents after saving to the camera roll?

If you don't need the file after saving it's best to delete it

> Though, I would have expected everything to be backwards compatible with minor version changes (7.4.0 works with no issues).

Looks like there was a breaking change then, not to my knowledge. But the package code is a bit messy, and some things are hard to test.
 
> Incidentally, we switched from `save` to `saveAsset` because the typescript typings say it's now [deprecated](https://github.com/react-native-cameraroll/react-native-cameraroll/blob/74fb19f1d7db1e4d5e1edd43c3e92fe51bf7ab45/src/CameraRoll.ts#L244). Wondering if this may also be related.

Well you can use it as long it works 😅

**Comment by gezquinndesign** [comment]
Saving to disk doesn't work either. Even with `Image.resolveAssetSource(localImage).uri`, where I know the `localImage` is a file that exists in the app, I get the same error.

**Comment by gezquinndesign** [comment]
@Arjan-Zuidema When I tried your download suggestion yesterday I failed to use the Documents directory (I was using the Download directory which, when using rn-fetch-blob, is [only available on android](https://github.com/joltup/rn-fetch-blob/wiki/File-System-Access-API#dirs)).  

Saving to the Documents directory before saving to the cameraroll resolves the issue. I'll close this issue and the PR.
